sonar-plugin-api/src/main/java/org/sonar/api/batch/fs/InputPath.java

```diff
@@ -34,26 +35,41 @@
   /**
    * @see InputFile#relativePath()
    * @see InputDir#relativePath()
+   * @deprecated since 6.5 use {@link #uri()}
    */
+  @Deprecated
   String relativePath();
 
   /**
    * @see InputFile#absolutePath()
    * @see InputDir#absolutePath()
+   * @deprecated since 6.5 use {@link #uri()}
    */
+  @Deprecated
   String absolutePath();
 
   /**
    * @see InputFile#file()
    * @see InputDir#file()
+   * @deprecated since 6.5 use {@link #uri()}
    */
+  @Deprecated
   File file();
 
   /**
    * @see InputFile#path()
    * @see InputDir#path()
    * @since 5.1
+   * @deprecated since 6.5 use {@link #uri()}
    */
+  @Deprecated
   Path path();
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)